### DISCUSSION TITLE
Does ImageMagick silently rasterize PDFs?

### DISCUSSION THREAD

**Discussion by johnmapeson** [ImageMagick/ImageMagick#7632] [Help] [answered]
### ImageMagick version

7.1.1-36

### Operating system, version and so on

macOS Sonoma 14.5

### Description

I often use ImageMagick to merge multiple PDFs into one, like this:

```
magick -density 300 input1.pdf input2.pdf output.pdf
```

But recently I have read:

> IM is a 'raster image processor', and while it can read or write images stored in one of the vector formats it does so by converting the image to and from an internal raster image.
>
> Consequently if you are trying to magick an image from a vector format, to another vector format, IM will essentially rasterize this image at the currently defined resolution or density which will hopefully (but unlikely) be suitable for the output device you intend to use it on.
>
> In other words, any output from IM will never be a true vector format. While it can magick its internal raster format into a vector format file, the result is only a superficial vector image wrapper around an image in raster format. And unless the raster image is defined properly (at the right resolution) for the output device, the result will not be particularly good.
>
> Unfortunately new uses to IM do not know anything about this. They see IM as a converter that can magick say PDF to Postscript, producing images with 'blocky' aliasing effects, 'washed out' colors, or blurry images that just do not look good at all, on the intended output device.
>
> http://www.imagemagick.org/Usage/formats/#vector

Does it mean that if I use the command like the one above, the text in resulting PDF had been rasterized?

**Comment by fmw42** [comment, marked as answer]
Yes, the final output PDF will be a raster image in a vector shell. So you will not be able to edit your text.  You may want to find another tool to combine multiple PDFs that preserve your text.

Also note to get higher quality, use a larger density and then resize before saving. For example, nominal density is 72. So 4x72=288. So use density 288 and then if you want the result smaller, resize by 1/4=25%. In you want the result larger, then ignore the resize and use any density larger than 72.
